pkg/util/log/logger.go

```diff
@@ -0,0 +1,344 @@
+package log
+
+import (
+	"fmt"
+
+	"github.com/elastic/beats/v7/libbeat/logp"
+	"github.com/sirupsen/logrus"
+)
+
+// FieldLogger Wraps the StdLogger, and provides logrus methods for logging with fields
+type FieldLogger interface {
+	StdLogger
+	WithField(key string, value interface{}) FieldLogger
+	WithFields(fields logrus.Fields) FieldLogger
+	WithError(err error) FieldLogger
+}
+
+// StdLogger interface for logging methods found in the go standard library logger.
+type StdLogger interface {
+	logrus.StdLogger
+
+	Debug(args ...interface{})
+	Debugf(format string, args ...interface{})
+	Debugln(args ...interface{})
+
+	Error(args ...interface{})
+	Errorf(format string, args ...interface{})
+	Errorln(args ...interface{})
+
+	Info(args ...interface{})
+	Infof(format string, args ...interface{})
+	Infoln(args ...interface{})
+
+	Warn(args ...interface{})
+	Warnf(format string, args ...interface{})
+	Warnln(args ...interface{})
+}
+
+// LogRedactor interface for redacting log messages
+type LogRedactor interface {
+	TraceRedacted(fields []string, args ...interface{})
+	ErrorRedacted(fields []string, args ...interface{})
+	InfoRedacted(fields []string, args ...interface{})
+	DebugRedacted(fields []string, args ...interface{})
+}
+
+// NewFieldLogger returns a FieldLogger for standard logging, and logp logging.
+func NewFieldLogger() FieldLogger {
+	entry := logrus.NewEntry(log)
+	return &logger{
+		entry: entry,
+	}
+}
+
+type logger struct {
+	entry *logrus.Entry
+}
+
+// WithField adds a field to the log message
+func (l *logger) WithField(key string, value interface{}) FieldLogger {
+	return &logger{entry: l.entry.WithField(key, value)}
+}
+
+// WithFields adds multiple fields to the log message
+func (l *logger) WithFields(fields logrus.Fields) FieldLogger {
+	return &logger{entry: l.entry.WithFields(fields)}
+}
+
+// WithError adds an error field to the message
+func (l *logger) WithError(err error) FieldLogger {
+	return &logger{entry: l.entry.WithError(err)}
+}
+
+// Debugf prints a formatted debug message
+func (l *logger) Debugf(format string, args ...interface{}) {
+	if l.isLogP() {
+		args = append(args, l.formatEntries()...)
+		logp.L().Debugf(format, args)
```

Comment:
@jcollins-axway The original log package uses a const called `debugSelector` when writing debug logs to logp. I don't know exactly what we are trying to do with that. Do you mind updating this PR with anything else we may need to keep it consistent with the current implementation?

Reply:
Logp does not have trace, so to differ the 2 logging events the debugSelector is used to name the logger in logp.